xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/widgets/gallery/gallery.js

```diff
@@ -133,11 +154,18 @@ XWiki.Gallery = Class.create({
     }
     // Update only if it's a different image. Some browsers, e.g. Chrome, don't fire the load event if the image URL
     // doesn't change. Another trick would be to reset the src attribute before setting the actual URL (set to '').
-    if (this.currentImage.src != this.images[index].url) {
+    const imageData = this.images[index];
+    if (this.currentImage.src !== imageData.url) {
       this.currentImage.style.visibility = 'hidden';
       Element.addClassName(this.currentImage.parentNode, 'loading');
-      this.currentImage.title = this.images[index].title;
-      this.currentImage.src = this.images[index].url;
+      this.currentImage.title = imageData.title;
+      // If the alt is just the name of the file, we instead fall back on the human-readable currentImage translation.
+      if (!imageData.url.includes(imageData.alt)) {
+        this.currentImage.alt = imageData.alt;
+      } else {
+        this.currentImage.alt = l10n['core.widgets.gallery.currentImage'];
```

Comment:
This doesn't seem like a good text alternative to me. At least from what I read in https://www.w3.org/WAI/WCAG21/Techniques/failures/F30 this is clearly a WCAG failure. To me, using the file name is slightly better as it could be meaningful.